server/src/main/scala/net/wiringbits/actions/UpdatePasswordAction.scala

```diff
@@ -0,0 +1,37 @@
+package net.wiringbits.actions
+
+import net.wiringbits.api.models.UpdatePassword
+import net.wiringbits.apis.EmailApi
+import net.wiringbits.apis.models.EmailRequest
+import net.wiringbits.repositories.UsersRepository
+import net.wiringbits.util.EmailMessage
+import org.mindrot.jbcrypt.BCrypt
+
+import java.util.UUID
+import javax.inject.Inject
+import scala.concurrent.{ExecutionContext, Future}
+
+class UpdatePasswordAction @Inject() (
+    usersRepository: UsersRepository,
+    emailApi: EmailApi
+)(implicit ec: ExecutionContext) {
+
+  def apply(userId: UUID, request: UpdatePassword.Request): Future[Unit] = {
+    val validate = Future {
+      if (request.newPassword.string.isEmpty) new RuntimeException(s"The password is required")
```

Comment:
Can it be empty? I think the model has validations to prevent constructing an empty password